### PR TITLE
use deterministic censusID

### PR DESCRIPTION
**Pull request by p4u**
Based on blockNumber and StrategyID we can derive
a unique identifier for a censusID.

This way we can easily check if the census already exist in the database.

### CODE REVIEW

File: api/censuses.go [hunk 1]
```diff
@@ -108,18 +109,42 @@ func (capi *census3API) launchCensusCreation(msg *api.APIdata, ctx *httprouter.H
 	return ctx.Send(res, api.HTTPstatusOK)
 }
 
+// innerCensusID generates a unique identifier by concatenating the BlockNumber, StrategyID,
+// and a numerical representation of the Anonymous flag from a CreateCensusRequest struct.
+// The BlockNumber and StrategyID are concatenated as they are, and the Anonymous flag is
+// represented as 1 for true and 0 for false. This concatenated string is then converted
+// to a uint64 to create a unique identifier.
+func innerCensusID(c *CreateCensusRequest) uint64 {
```
Comment (lucasmenendez) [inline api/censuses.go:117]: I think that this function could be moved to `census/census.go`

**Comment by p4u** [comment]
@CodiumAI-Agent /review

**Comment by p4u** [comment]
@CodiumAI-Agent /improve --extended

**Comment by p4u** [comment]
@CodiumAI-Agent /describe

**Pull request merged**